### PR TITLE
🩹  added patch command for subworkflows

### PR DESCRIPTION
**Pull request by ctuni**
Related to issue: https://github.com/nf-core/tools/issues/1891

Is there a straightforward way to test this?

### CODE REVIEW

**Comment by mashehu** [comment]
for testing you can do similar things to https://github.com/nf-core/tools/blob/master/tests/modules/patch.py

**Comment by mirpedrol** [comment]
Hello! @ctuni are you planning to work on this PR, otherwise I would like to take over and finish it up :)

**Comment by ctuni** [comment]
Hello! Thank you for reviving this, please go ahead and let me know if I can help with anything!

**Comment by ewels** [comment]
@ctuni this might help your future development work 😉 

```bash
pip install pre-commit
pre-commit install
```

It'll run `ruff` and other tests as part of `git commit` and fix formatting errors before you push..

File: nf_core/__main__.py [hunk 1]
```diff
@@ -1567,6 +1567,43 @@ def command_subworkflows_install(ctx, subworkflow, directory, prompt, force, sha
...
+@click.pass_context
+@click.argument("tool", type=str, required=False, metavar="<tool> or <tool/subtool>")
+@click.option(
+    "-d",
+    "--dir",
+    type=click.Path(exists=True),
+    default=".",
+    help=r"Pipeline directory. [dim]\[default: current working directory][/]",
+)
+@click.option("-r", "--remove", is_flag=True, default=False)
```
Comment (mashehu) [inline nf_core/__main__.py:1581]: help text missing

File: nf_core/subworkflows/lint/subworkflow_changes.py [hunk 2]
```diff
@@ -20,7 +23,29 @@ def subworkflow_changes(subworkflow_lint_object, subworkflow):
...
+                subworkflow.org,
+                subworkflow.patch_path,
+                tempdir,
+                reverse=True,
+            )
+            for file, lines in new_lines.items():
+                with open(tempdir / file, "w") as fh:
+                    fh.writelines(lines)
+        except LookupError:
+            # This error is already reported by subworkflow_patch, so just return
```
Comment (mashehu) [inline nf_core/subworkflows/lint/subworkflow_changes.py:45]: wtihout checking, this feels like we would also use the same logic inn `modules_changes`. can we make a `components_changes` superfunction?

File: tests/subworkflows/test_patch.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        self.setup_patch(self.pipeline_dir, True)
+
+        # Try creating a patch file
+        patch_obj = nf_core.subworkflows.SubworkflowPatch(self.pipeline_dir, GITLAB_URL, GITLAB_SUBWORKFLOWS_BRANCH)
+        patch_obj.patch("bam_sort_stats_samtools")
+
+        subworkflow_path = Path(self.pipeline_dir, "subworkflows", GITLAB_REPO, "bam_sort_stats_samtools")
+
+        # Check that a patch file with the correct name has been created
+        assert set(os.listdir(subworkflow_path)) == {"main.nf", "meta.yml", "bam_sort_stats_samtools.diff"}
```
Comment (mashehu) [inline tests/subworkflows/test_patch.py:82]: less error prone if we only check that `bam_sort_stats_samtools.diff` is in the directory

**Review by mashehu — APPROVED** [review]
Changelog entry is missing

File: tests/modules/test_patch.py [hunk 4]
```diff
@@ -76,7 +76,7 @@ def test_create_patch_no_change(self):
         module_path = Path(self.pipeline_dir, "modules", REPO_NAME, BISMARK_ALIGN)
 
         # Check that no patch file has been added to the directory
-        assert set(os.listdir(module_path)) == {"main.nf", "meta.yml", "environment.yml"}
+        assert "bismark-align.diff" in set(os.listdir(module_path))
```
Comment (mashehu) [inline tests/modules/test_patch.py:79]: this is not the same assertion as was there before 🤔

Comment (mirpedrol) [reply tests/modules/test_patch.py:79]:
```suggestion
        assert "bismark-align.diff" not in set(os.listdir(module_path))
```

File: tests/modules/test_patch.py [hunk 5]
```diff
@@ -94,7 +94,7 @@ def test_create_patch_change(self):
 
         patch_fn = f"{'-'.join(BISMARK_ALIGN.split('/'))}.diff"
         # Check that a patch file with the correct name has been created
-        assert set(os.listdir(module_path)) == {"main.nf", "meta.yml", "environment.yml", patch_fn}
+        assert patch_fn in set(os.listdir(module_path))
```
Comment (mashehu) [inline tests/modules/test_patch.py:97]: how about we test directly the file existence:
```suggestion
       assert Path(module_path) / patch_fn.exists()
```
same for the others

Comment (mirpedrol) [reply tests/modules/test_patch.py:97]: makes sense, thanks for the suggestion 😄

**Pull request merged**